### ISSUE TITLE
Can't enforce validity window despite consensus

### ISSUE THREAD

**Issue by Eligioo** [nimiq/core-rs-albatross#1703]
Some nodes fail to enforce the validity window with results into the validator never can become active.

In https://github.com/nimiq/core-rs-albatross/blob/albatross/validator/src/validator.rs#L799
`can_be_active` always returns false because the `can_enforce_validity_window` always returns false.

Note the extra logs added which output the values of `can_enforce_validity_window` while the node is running.
```
2023-07-13T08:39:47.197154000Z INFO  wrappers             | can_enforce_validity_window: history_store.length_at: 0, first window: 210192, election block: 216000, block number 217391
2023-07-13T08:39:47.197175000Z INFO  validator            | blockchain event event=Extended(4dde618aef0c15eab1042b9db658c9605866083ac3309f5056fb1c166dad4741) can_be_active=false
```

[Log_Length_at_always_0_despite_consensus.log](https://github.com/nimiq/core-rs-albatross/files/12038168/Log_Length_at_always_0_despite_consensus.log)

**Comment by paberr** [comment]
@Eligioo has this been fixed by #1707 ?

**Comment by Eligioo** [comment]
Yes, after monitoring my node since this PR landed, it hasn't occurred anymore.
